File: 015.c
#include <stdio.h>

unsigned long n_paths = 0;
void move_point(int x, int y, int xmax, int ymax) {
  if (x == xmax && y == ymax) {
    n_paths++;
  } else if (y < ymax) {
    move_point(x, y + 1, xmax, ymax);
    if (x < xmax) {
      move_point(x + 1, y, xmax, ymax);
    }
  } else if (x < xmax) {
    move_point(x + 1, y, xmax, ymax);
  }
}

int main() {
  move_point(0, 0, 20, 20);
  printf("n_paths: %lu\n", n_paths);
  return (0);
}

File: 015.py
nPaths = 0;
def move_point(point, limit):
    global nPaths
    if point == limit:
        #print("Finished!")
        nPaths += 1
    if point[1] < limit[1]:
        # Continue down...
        new_point = (point[0], point[1]+1)
        # print("From", point, "to", new_point)
        move_point(new_point, limit)
        # And lets go right too
        if point[0] < limit[0]:
            new_point = (point[0]+1, point[1])
            # print("From", point, "to", new_point)
            move_point(new_point, limit)
    elif point[0] < limit[0]:
        new_point = (point[0]+1, point[1])
        # print("From", point, "to", new_point)
        move_point(new_point, limit)

move_point((0, 0), (20, 20))
print("N Paths", nPaths)
